tools/hyperistio/hyperistio.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/mixer/test/client/env"
+	"istio.io/istio/pilot/pkg/bootstrap"
+	"istio.io/istio/pilot/pkg/model"
+	envoy "istio.io/istio/pilot/pkg/proxy/envoy/v1"
+	"istio.io/istio/pilot/pkg/serviceregistry"
+	agent "istio.io/istio/pkg/bootstrap"
+	"istio.io/istio/tests/util"
+)
+
+// hyperistio runs all istio components in one binary, using a directory based config by
+// default. It is intended for testing/debugging/prototyping.
+func main() {
+	flag.Parse()
+	err := startAll()
+	if err != nil {
+		log.Fatal("Failed to start ", err)
+	}
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	<-sigs
+
+	//select{}
+}
+
+func startAll() error {
+	err := startPilot()
+	if err != nil {
+		return err
+	}
+
+	err = startMixer()
+	if err != nil {
+		return err
+	}
+
+	// Mixer test servers
+	srv, err := env.NewHTTPServer(7070)
+	if err != nil {
+		return err
+	}
+	srv.Start()
+
+	go util.RunHTTP(7072, "v1")
+	go util.RunGRPC(7073, "v1", "", "")
+	go util.RunHTTP(7074, "v2")
+	go util.RunGRPC(7075, "v2", "", "")
+	err = startEnvoy()
```

Comment:
wont you need two envoys? client side and server side

Reply:
Right, one step at a time. 
Probably 3 - ingress and 2 sidecars.